devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -239,6 +260,34 @@ public void execute() throws MojoFailureException, MojoExecutionException {
             }
             devModeContext.getBuildSystemProperties().putAll((Map) project.getProperties());
             devModeContext.setSourceEncoding(getSourceEncoding());
+            devModeContext.setSourceJavaVersion(source);
```

Comment:
Actually I don't think it should be called `sourceJavaVersion` since it should (in a future PR) be able to handle Kotlin and Scala as well, right?

Reply:
I don't know whether the Kotlin and Scala compilers have equivalent flags and whether their Maven plugins have equivalent settings.  If not, then I would argue that the current name is more accurate.  Do you know whether Scala and Kotlin users would expect `source` and `target` settings to exist, and what they would mean?  I'm pretty sure that for Scala, using a different source version would mean switching to a different version of the compiler.  I'm also not sure what the `target` setting would mean in that context.

Reply:
I don't know anything about Scala. However with Kotlin you can definitely set the `target` version, see https://kotlinlang.org/docs/reference/using-maven.html#attributes-specific-for-jvm

Reply:
OK, renamed `targetJavaVersion` to `targetJvmVersion`.

Reply:
As for the source version, do we really want a single variable that covers both languages?  That might cause trouble when you're mixing several languages in the same project.  It might make more sense to rename the `<source/>` element to `<javaSource>` in the plugin configuration instead.  What do you think?

Reply:
Currently mixing several languages will (most likely) not work in Dev mode because there is way to setup the proper order of compilation.
We should probably just keep source for now I think.